js/modifyContributionForm.js

```diff
@@ -0,0 +1,83 @@
+CRM.$(function ($) {
+
+  (function() {
+    setTotalAmount();
+    hideStatusField();
+    setAmountCurencySymbol();
+    toggleRecordPaymentBlock();
+    placePaymentFieldsTogether();
+  })();
+
+  function hideStatusField() {
+    CRM.$('.crm-contribution-form-block-contribution_status_id').hide();
+  }
+
+  function toggleRecordPaymentBlock() {
+    const recordPaymentCheck = document.querySelector("input[name=fe_record_payment_check]");
+    const toggle = (checked)  => {
+      if (checked) {
+        $('.record_payment-block').show();
+      } else {
+        $('.record_payment-block').hide();
+      }
+    }
+
+    toggle(recordPaymentCheck.checked)
+    recordPaymentCheck.addEventListener('change', function() {
+      toggle(this.checked)
+    });
+  }
+
+  function setAmountCurencySymbol() {
+    const setSymbol = () => {
+      const currencySelect = $('#currency').val();
+      const currencySymbol = CRM.vars.financeextras.currencies[currencySelect];
+
+      $('.record_payment-block #currency-symbol').text(currencySymbol)
+    };
+  
+    setSymbol();
+    $('select[name=currency]').on('change', function() {
+      setSymbol();
+    });
+  }
+
+  function setTotalAmount() {
+    const recordPaymentAmount = document.querySelector("input[name=fe_record_payment_amount]");
+    $('#total_amount').on("change", function() {
```

Comment:
does this work with the line item editor extension?

Reply:
no, it doesn't, I left a comment on the Spec,

> To ensure this behaviour for contribution with line items, we would need to have a duplicate of the total calculation code used by the line_item editor extension in our extension.
> 
> of which I don't think is necessary, I believe we can exempt this.

I'm saying this because we would need to watch each change on the line items and calculate their tax value from the financial type etc.  The real problem is that the line item editor extension doesn't expose the total amount.

https://github.com/compucorp/lineitemedit/blob/8578f6f8343d6eea4e8df99a11ead16d010e8f6e/templates/CRM/Lineitemedit/Form/AddLineItems.tpl#L145-L180

what do you think @omarabuhussein ?

Reply:
@olayiwola-compucorp while I don't have a problem with not supporting line item editor and that the user should alter the payment amount manually, but given that it is more of a product usability problem, I will leave this for @jamienovick to decide.